js/build.js

```diff
@@ -555,11 +555,17 @@ Fliplet.Widget.instance('login', function(data) {
             return Promise.reject(T('widgets.login.fliplet.warnings.noRedirectWhenEditing'));
           }
 
-          var navigate = Fliplet.Navigate.to(_this.data.action);
+          return Fliplet.Session.get().then(function(session) {
+            if (session.client && session.client.source === 'studio') {
+              return Promise.reject('Logout prevented in preview mode.');
```

Comment:
@armine-fliplet This reject error is confusing in a sense because when a user is logging out in Preview mode, we are destroying session passport from backend which is equivalent to logging out. Error message should be something like: 'Preventing navigation to another screen in Preview mode.'